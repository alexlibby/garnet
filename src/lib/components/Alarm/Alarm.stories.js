import Alarm from "./Alarm.svelte";
import { fn } from "@storybook/test";
import { action } from "@storybook/addon-actions";
import { BADGE } from "storybook-addon-badges";

const BADGES = {
  ...BADGE,
  ALPHA: "Alpha",
};

export default {
  title: "Garnet UI Library/Animation Components/Alarm",
  component: Alarm,
  argTypes: {
    disabled: { control: "boolean" },
    label: { control: "string" },
    small: { control: "boolean" },
    notifications: { control: "integer" },
  },
  props: {
    label: "Alarm with label",
    disabled: false,
    filled: false,
  },
  on: { change: fn().mockName("on-change") },
};

export const Default = (args) => ({
  Component: Alarm,
  props: {
    ...args,
    notifications: 23,
  },
  on: { change: fn().mockName("on-change") },
});

Default.parameters = {
  badges: [BADGES.ALPHA],
};

export const Filled = (args) => ({
  Component: Alarm,
  props: {
    ...args,
    notifications: 12,
    filled: true,
  },
  on: { change: fn().mockName("on-change") },
});

Filled.parameters = {
  badges: [BADGES.ALPHA],
};

export const Small = (args) => ({
  Component: Alarm,
  props: {
    ...args,
    filled: true,
    small: true,
  },
  on: { change: fn().mockName("on-change") },
});

Small.parameters = {
  badges: [BADGES.ALPHA],
};
