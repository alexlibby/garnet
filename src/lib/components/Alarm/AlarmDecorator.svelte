<script>
  // placeholder - value not used
  let notifications = 1;

  function addNotification() {
    const prevNum = Number(document.querySelector(".countNum").innerText);
    notifications = prevNum + 1
    document.querySelector(".countNum").textContent = notifications;
  }
</script>

<p>Press button to increase the notification count:</p>
<button on:click={addNotification}>âž•</button>
<button on:click={() => notifications = 0}>Reset</button>

<div>
  <slot />
</div>

