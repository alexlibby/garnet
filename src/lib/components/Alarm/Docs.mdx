import { Canvas, Meta } from "@storybook/blocks";
import { ArgTypes } from "@storybook/blocks";
import * as AlarmStories from "./Alarm.stories";

<Meta of={AlarmStories} />

# Alarm

Alarm is the primary component. It has 3 possible states.

- [Default](#default)
- [Filled](#filled)
- [Small](#small)

## Default

This is the default state for our Alarm component:

<Canvas of={AlarmStories.Default} />

## Filled

This state shows the Alarm component with a filled icon:

<Canvas of={AlarmStories.Filled} />

## Small

This state shows a smaller version of the Alarm component:

<Canvas of={AlarmStories.Small} />

## Properties of component

Below is a list of arguments available for this component:

<ArgTypes of={AlarmStories} />
