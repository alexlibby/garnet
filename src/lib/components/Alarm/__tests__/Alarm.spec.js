import { describe, test, expect } from "vitest";
import { axe } from "vitest-axe";
import * as AxeMatchers from "vitest-axe/matchers";
import { render, screen, fireEvent } from "@testing-library/svelte";
import Alarm from "../../Alarm/Alarm.svelte";

const AlarmHTML = `<div class="garnet-alarm s-xm1t82NY3YZU"><h3 class="s-xm1t82NY3YZU">This is a test label</h3> <div class="notifications s-xm1t82NY3YZU"><div class="count  s-xm1t82NY3YZU" style="--countColor: #733635; --countBackgroundColor: #f4f4f4;"><span data-testid="notifications" class="s-xm1t82NY3YZU">23</span></div> <svg class="bell  outline ring s-xm1t82NY3YZU" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor"><path d="M14.857 17.082a23.848 23.848 0 005.454-1.31A8.967 8.967 0 0118 9.75v-.7V9A6 6 0 006 9v.75a8.967 8.967 0 01-2.312 6.022c1.733.64 3.56 1.085 5.455 1.31m5.714 0a24.255 24.255 0 01-5.714 0m5.714 0a3 3 0 11-5.714 0" stroke-linecap="round" stroke-linejoin="round" class="s-xm1t82NY3YZU"></path></svg></div></div>`;
expect.extend(AxeMatchers);

describe("Tests for Alarm", () => {
  const props = {
    label: "Alarm with label",
    notifications: 23,
    filled: false,
  };

  test("should render properly", () => {
    const result = render(Alarm, props);
    expect(() => result).not.toThrow();
  });

  test("should show correct number of notifications", () => {
    const result = render(Alarm, props);

    const notifications = result.getByTestId("notifications").textContent;
    expect(notifications).toEqual("23");
  });

  test("get a snapshot of component", () => {
    // Create a new container for the test and mount component
    const host = document.createElement("div");
    document.body.appendChild(host);
    const instance = new Alarm({ target: host, props: props });

    // check that all rendered OK
    expect(instance).toBeTruthy();

    // Take snapshot
    expect(host.innerHTML).toMatchSnapshot();
  });

  test("should demonstrate no issues with accessibility", async () => {
    const render = () => AlarmHTML;
    const result = render();

    const results = await axe(result, {
      rules: {
        region: { enabled: false },
      },
    });

    expect(results).toHaveNoViolations();
  });
});
