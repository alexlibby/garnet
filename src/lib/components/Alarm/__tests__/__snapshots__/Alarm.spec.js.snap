// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests for Alarm > get a snapshot of component 1`] = `"<p class="svelte-iolp12">Press button to increase the notification count:</p> <button>âž•</button> <button>Reset</button> <div class="garnet-alarm"><h3 class="svelte-iolp12">Alarm with label</h3> <div class="notifications svelte-iolp12"><div class="count  svelte-iolp12" style="--countColor: #733635; --countBackgroundColor: #f4f4f4;"><span data-testid="notifications">23</span></div> <svg class="bell  outline ring svelte-iolp12" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor"><path d="M14.857 17.082a23.848 23.848 0 005.454-1.31A8.967 8.967 0 0118 9.75v-.7V9A6 6 0 006 9v.75a8.967 8.967 0 01-2.312 6.022c1.733.64 3.56 1.085 5.455 1.31m5.714 0a24.255 24.255 0 01-5.714 0m5.714 0a3 3 0 11-5.714 0" stroke-linecap="round" stroke-linejoin="round"></path></svg></div></div>"`;
